sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_responses.hpp

```diff
@@ -9,15 +9,49 @@
 
 namespace Azure { namespace Storage { namespace Files { namespace Shares { namespace Models {
 
+  struct FileShareSmbProperties
+  {
+    /**
+     * @brief Key of the permission to be set for the directory/file.
+     */
+    Azure::Core::Nullable<std::string> PermissionKey;
+
+    /**
+     * @brief If specified, the provided file attributes shall be set. Default value:
+     * FileAttribute::Archive for file and FileAttribute::Directory for directory.
+     * FileAttribute::None can also be specified as default.
+     */
+    FileAttributes Attributes = static_cast<FileAttributes>(0);
+
+    /**
+     * @brief Creation time for the file/directory..
+     */
+    Azure::Core::Nullable<Core::DateTime> CreatedOn;
+
+    /**
+     * @brief Last write time for the file/directory..
+     */
+    Azure::Core::Nullable<Core::DateTime> LastWrittenOn;
+  };
+
   // ServiceClient models:
 
   using ListSharesSinglePageResult = ServiceListSharesSinglePageResult;
   using SetServicePropertiesResult = ServiceSetPropertiesResult;
   using GetServicePropertiesResult = StorageServiceProperties;
 
   // ShareClient models:
-  using CreateShareResult = ShareCreateResult;
-  using DeleteShareResult = ShareDeleteResult;
```

Comment:
I got a questions here. `ShareCreateResult` is in `Models` namespace. But it looks to me is an implementation detail and shouldn't be exposed to our customers. So should we move these return types to `Details` namespace?

Reply:
Yes, but due to the `single-purpose` limitation, I think I won't address this issue in this PR, feel free to open an issue to track it.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/1287